ls1intum/Artemis
# `General`: Fix an issue during user deletion with existing conversations

### Checklist
#### General


#### Server

### Motivation and Context
Closes #6365 

### Description

Modified the `deleteUser(User user)` function of `UserService` to also remove the following from corresponding repositories:
- **Posts** - belonging to the user, to Conversations created by the user, to OneToOneChats where the user is a participant
- **OneToOneChats** - where the user is a participant
- **Conversations** - created by the user
- **ConversationParticipants** - created by the user

Also added necessary tests to ensure the deletion of the aforementioned objects when a mock user is deleted.

### Steps for Testing

Prerequisites:
- 1 Admin
- 2 Users (any role)
- 1 Course

1. **Log in as Admin**
2. Navigate to the Messages section of a Course
3. Create a channel that the Users can join
4. **Log in as the User that will be deleted**
5. Register to the course (if not already registered)
6. Navigate to the Messages section of the Course
7. Send a direct message to another user
8. Send a message to a group
9. Send a message to a channel
10. **Log in as the 2nd user**
11. Send a direct message to the user that will be deleted
12. **Log in as Admin**
13. Delete the user (should succeed)

### Review Progress






#### Code Review
#### Manual Tests
